hil-test/tests/delay_async.rs

```diff
@@ -0,0 +1,77 @@
+//! Async Delay Test
+//!
+//! Specifically tests the various implementations of the
+//! `embedded_hal_async::delay::DelayNs` trait.
+
+//% CHIPS: esp32 esp32c2 esp32c3 esp32c6 esp32h2 esp32s2 esp32s3
+
+#![no_std]
+#![no_main]
+
+use embedded_hal_async::delay::DelayNs;
+#[cfg(systimer)]
+use esp_hal::timer::systimer::{Alarm, FrozenUnit, SystemTimer};
+use esp_hal::{peripherals::Peripherals, timer::timg::TimerGroup};
+use hil_test as _;
+
+struct Context {
+    peripherals: Peripherals,
+}
+
+async fn test_async_delay_ns(mut timer: impl DelayNs, duration: u32) {
+    let t1 = esp_hal::time::now();
+    timer.delay_ns(duration).await;
```

Comment:
You should also test that the other delay fns are working in the correct unit (i.e. that delay_ms(X) doesn't wait for X us accidentally)